### PR TITLE
PLF-6637 : display only allowed groups for user

### PR DESCRIPTION
**Pull request by thomasdelhomenie**
The group and membership selector WebUI components listed all the platform groups, no matter the user permissions.

This fix makes org.exoplatform.social.webui.UISocialGroupSelector depends on the Gatein component org.exoplatform.webui.organization.account.UIGroupSelector.UIGroupSelector in order to reuse the same logic, meaning filtering groups and listing a group only if:
* the user is the super user
* the user is a platform administrator
* the user is in the group

### CODE REVIEW

File: component/webui/src/main/java/org/exoplatform/social/webui/UISocialGroupSelector.java [hunk 1]
```diff
@@ -17,342 +17,69 @@
...
-      return list;
-    OrganizationService service = getApplicationComponent(OrganizationService.class);
-    Group group = service.getGroupHandler().findGroupById(id);
-    if (group == null)
-      return list;
-    list.add(0, new LocalPath(group.getId(), group.getGroupName()));
-    getPath(list, group.getParentId());
-    return list;
+  @Override
+  protected boolean canUserSeeGroup(Group group) {
```
Comment (boubaker) [inline component/webui/src/main/java/org/exoplatform/social/webui/UISocialGroupSelector.java]: Same remark as https://github.com/exodev/ecms/pull/469#pullrequestreview-237040984.
BTW, why redefining this behavior (no supposed changes comparing to gateIn component behavior according to PR description)

Comment (thomasdelhomenie) [reply component/webui/src/main/java/org/exoplatform/social/webui/UISocialGroupSelector.java]: The PRs are not done yet, it is still a work in progress, waiting for functional rules (https://jira.exoplatform.org/browse/PLF-6637?focusedCommentId=499496&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-499496).

File: component/webui/src/main/java/org/exoplatform/social/webui/SpaceCreationInvitationGroupVisibilityPlugin.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import org.exoplatform.services.security.ConversationState;
+import org.exoplatform.services.security.Identity;
+import org.exoplatform.services.security.MembershipEntry;
+import org.exoplatform.social.core.space.SpacesAdministrationService;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SpaceCreationInvitationGroupVisibilityPlugin extends GroupVisibilityPlugin {
```
Comment (boubaker) [inline component/webui/src/main/java/org/exoplatform/social/webui/SpaceCreationInvitationGroupVisibilityPlugin.java]: could you please add javadoc and copyright comment

Comment (thomasdelhomenie) [reply component/webui/src/main/java/org/exoplatform/social/webui/SpaceCreationInvitationGroupVisibilityPlugin.java]: Done

**Comment by thomasdelhomenie** [comment]
@boubaker @wkhessairi I noticed that the webui permission selector is not used anymore to invite users in a space. Shouldn't we remove it ?

**Comment by boubaker** [comment]
> @boubaker @wkhessairi I noticed that the webui permission selector is not used anymore to invite users in a space. Shouldn't we remove it ?

Yes, its usage has been removed (functional point of view) in second iteration of Space Templates. You can keep changes on this PR to keep this algorithm of group permissions selection, we will treat the code cleanup before merging the third iteration of feature by deleting `org.exoplatform.social.webui.UIUsersInGroupSelector` and `org.exoplatform.social.webui.UISocialGroupSelector`.

**Comment by thomasdelhomenie** [comment]
ok
Actually I was thinking that the changes on webui selector will be still needed when backporting it on a maintenance version.

**Comment by boubaker** [comment]
Yep, if it will be backported to 5.2.x branch, we will need to keep it as it is and avoid making the deletion of component in this issue too.

**Pull request merged**